docs/adr/0003-event-broker-for-subscription-platform.md

```diff
@@ -0,0 +1,108 @@
+    * C. Storing webhook and login activity in FxA Event Broker
+
+## Decision Outcome
+
+Chosen Option: C. Implementing RP notification as a new FxA Event Broker service
+with webhook and login activity stored in FxA Event Broker, because
+
+* Less subscription platform timeline risk to store data in new database vs.
+  modify existing FxA OAuth database.
+* Storing events at scale has database requirements that don't fit in well with
+  the limitations with MySQL encountered in FxA Auth/OAuth.
+* Having FxA Auth be the only store of which RP to notify would require each
+  notification to also include what RPs to notify, increasing the load on the
+  FxA Auth database.
+
+## Pros and Cons of the Options
+
+### A. Implementing RP notification as a FxA Auth Service feature w/webhook & login activity
+
+* Description
+  * FxA Auth Service has additional feature code added to perform webhook
+    deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+* Pros
+  * Auth service has direct access to subscription capability mapping and
+    OAuth webhook URLs.
+* Cons
+  * Event logs for each user are not well suited for existing MySQL database.
+  * Additional database schema migrations needed.
+  * Later features needing to query for user events could impact FxA Auth Service
+    availability.
+
+### B. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Auth Service
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+  * FxA Auth Server includes RPs that should be notified in the relevant
+    Service Notifications.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
+* Pros
+  * Division of event storing and distribution to a service that can be built
+    specifically for that function.
+* Cons
+  * Queries to FxA Auth Service required for each message to determine whether
+    to deliver to an RP could impact FxA Auth Service availability.
+  * FxA OAuth server should still have a record of which RPs a user has logged
+    into so a later issue to add this is still needed.
+
+### C. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Event Broker
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
+  * FxA Event Broker logs events and can determine which RPs a user has logged
+    into without any external service queries.
+* Pros
+  * Division of event storing and distribution to a service that can be built
+    specifically for that function.
+* Cons
+  * FxA OAuth server should still have a record of which RPs a user has logged
+    into so a later issue to add this is still needed.
```

Comment:
IIUC, we plan to let is obtain such a list of querying the event broker service, might be worth mentioning that here as a nod to the "built with future extensibility in mind" point from earlier in the doc.

Reply:
I'm not sure how to phrase that. Any suggestions?

Reply:
Maybe something like: "One possible approach would be to treat the event broker as the canonical source for this data, and have OAuth server query it on demand".

It's not super critical either way :-)

Reply:
Yea, at the moment we don't want the additional load on FxA OAuth, so definitely something to revisit later perhaps if/when FxA Auth/OAuth uses a different db backend.